File: src/main/java/com/github/weewar/mapviewer/dao/WeewarMapDAO.java
package com.github.weewar.mapviewer.dao;

import com.github.weewar.mapviewer.model.MapHeader;
import com.github.weewar.mapviewer.model.WeewarMap;

import java.util.List;
import java.util.Optional;

public interface WeewarMapDAO {
    Optional<WeewarMap> findMapById(int mapId);

    Optional<MapHeader> findMapHeaderById(int mapId);

    List<MapHeader> findMapHeaders(MapSearchCriteria searchCriteria);
}

File: src/main/kotlin/com/github/weewar/mapviewer/dao/WeewarMapDAO.kt
package com.github.weewar.mapviewer.dao

import com.github.weewar.mapviewer.model.MapHeader
import com.github.weewar.mapviewer.model.WeewarMap
import java.util.*

interface WeewarMapDAO {
    fun findMapById(mapId: Int): Optional<WeewarMap>

    fun findMapHeaderById(mapId: Int): Optional<MapHeader>

    fun findMapHeaders(searchCriteria: MapSearchCriteria): List<MapHeader>
}
